### PR TITLE
Mount volume if exists before starting grpc service

### LINKED ISSUE #469: csi provisioner pod lazy mount issue
csi provisioner pod mounts Kadalu Storages when it receives the PVC. Once mounted it keeps the mount for future use.

Quota crawler(https://github.com/kadalu/kadalu/blob/devel/csi/quota-crawler.sh#L3) and Metrics exporter(https://github.com/kadalu/kadalu/blob/devel/csi/exporter.py#L48) depends on these mount point to expose the data.

If Provisioner pods restarts then it will not mount all the storages unless it receives a new PVC again.

Solution: Mount all the available Kadalu Storages when Provisioner pod starts.

### PR DESCRIPTION
**Pull request by leelavg**
Fixes: https://github.com/kadalu/kadalu/issues/469

Signed-off-by: Leela Venkaiah G <lgangava@redhat.com>

### CODE REVIEW

**Comment by leelavg** [comment]
Diff in import is due to `isort`, will add notes in README for linting and fixing python files after a discussion

-----------Status--------------------------
```
# k get pods
NAME                          READY   STATUS    RESTARTS   AGE
operator-88bd4784c-2rhvn      1/1     Running   0          50m
kadalu-csi-nodeplugin-vk9jh   3/3     Running   0          50m
kadalu-csi-nodeplugin-pct2w   3/3     Running   0          50m
kadalu-csi-provisioner-0      5/5     Running   0          50m
kadalu-csi-nodeplugin-xvqwz   3/3     Running   0          50m
kadalu-csi-nodeplugin-p5h6s   3/3     Running   0          50m
server-replica3-2-0           1/1     Running   0          14m
server-replica3-1-0           1/1     Running   0          14m
server-replica3-0-0           1/1     Running   0          14m

# k get sts
NAME                     READY   AGE
kadalu-csi-provisioner   1/1     50m
server-replica3-2        1/1     14m
server-replica3-1        1/1     14m
server-replica3-0        1/1     14m

# k exec -it kadalu-csi-provisioner-0 -c kadalu-provisioner -- sh -c 'ps -ef | grep gluster'
root         928       1  0 05:41 ?        00:00:01 /opt/sbin/glusterfs --process-name fuse -l /var/log/gluster/gluster.log --volfile-id replica3 --fs-display-name kadalu:replica3 -f /kadalu/volfiles/replica3.client.vol /mnt/replica3 --client-pid -14
root        2743       0  0 05:56 pts/0    00:00:00 sh -c ps -ef | grep gluster
root        2752    2743  0 05:56 pts/0    00:00:00 grep gluster
```

------------Pod is deleted (Mount didnt' happen)--------------------
```
# k delete pod kadalu-csi-provisioner-0
pod "kadalu-csi-provisioner-0" deleted

# k exec -it kadalu-csi-provisioner-0 -c kadalu-provisioner -- sh -c 'ps -ef | grep gluster'
root          30       0  0 05:58 pts/0    00:00:00 sh -c ps -ef | grep gluster
root          38      30  0 05:58 pts/0    00:00:00 grep gluster
```
-----------With this PR (Mount happens on restart)
```
# k cp csi/main.py kadalu-csi-provisioner-0:/kadalu/main.py -c kadalu-provisioner

# k exec -it kadalu-csi-provisioner-0 -c kadalu-provisioner -- sh -c 'grep mount_storage /kadalu/main.py'
def mount_storage():                                                                                    
    mount_storage()

# k exec -it kadalu-csi-provisioner-0 -c kadalu-provisioner -- sh -c 'ps -ef | grep python'
root           1       0  0 06:26 ?        00:00:00 python3 /kadalu/start.py
root           9       1  0 06:26 ?        00:00:00 python3 /kadalu/main.py
root          10       1  0 06:26 ?        00:00:00 python3 /kadalu/exporter.py
root         103       0  0 06:28 pts/0    00:00:00 sh -c ps -ef | grep python
root         111     103  0 06:28 pts/0    00:00:00 grep python

# k exec -it kadalu-csi-provisioner-0 -c kadalu-provisioner -- sh -c 'kill 9'

# k exec -it kadalu-csi-provisioner-0 -c kadalu-provisioner -- sh -c 'ps -ef | grep gluster'
root         175       1  0 06:30 ?        00:00:00 /opt/sbin/glusterfs --process-name fuse -l /var/log/gluster/gluster.log --volfile-id replica3 --fs-display-name kadalu:replica3 -f /kadalu/volfiles/replica3.client.vol /mnt/replica3 --client-pid -14
root         223       0  0 06:30 pts/0    00:00:00 sh -c ps -ef | grep gluster
root         231     223  0 06:30 pts/0    00:00:00 grep gluster

# When attached to a TTY of provisioner
# k attach sts/kadalu-csi-provisioner -c kadalu-provisioner -it
Unable to use a TTY - container kadalu-provisioner did not allocate one
If you don't see a command prompt, try pressing enter.
[2021-03-12 06:30:09,161] INFO [kadalulib - 369:monitor_proc] - Restarted Process        name=csi
[2021-03-12 06:30:09,588] INFO [main - 36:mount_storage] - replica3 is mounted successfully
[2021-03-12 06:30:09,606] INFO [main - 56:main] - Server started
```
----------------------------------------------------------------------------

File: csi/main.py [hunk 1]
```diff
@@ -1,22 +1,41 @@
...
+
+    host_volumes = get_pv_hosting_volumes({})
+    for volume in host_volumes:
+        hvol = volume['name']
+        mntdir = os.path.join(HOSTVOL_MOUNTDIR, hvol)
+        try:
+            mount_glusterfs(volume, mntdir)
+        except CommandException:
+            logging.error("Unable to mount volume %s" % hvol)
+        logging.info("%s is mounted successfully" % hvol)
```
Comment (amarts) [inline csi/main.py]: looking at the pylint error, and to keep consistency with other places, best is to do it as below:

```suggestion
        logging.info(logf("Mounted successfully" , hvol=hvol))
```

same for above. (note the required `from kadalulib import logf`)

**Review by amarts — COMMENTED** [review]
LGTM

**Comment by leelavg** [comment]
@amarts, @aravindavk Can you please state any linters/fixers that you are using locally before pushing to github.

The idea is to have .pylintrc at the root of repo and use that when pushing to github and in CI as well, below are the tools that I use before pushing to remote. IMHO it'll be helpful to maintain consistency among all python src in the repo.

```
# type pyc
pyc is a function
pyc () 
{ 
    printf "\n%10s Sorting imports by 'isort'\n";
    isort "$@";
    printf "\n%10s Running through 'yapf'\n";
    yapf -i "$@";
    printf "\n%10s Remove unused vars/imports by 'autoflake'\n";
    autoflake --in-place --remove-unused-variables --remove-all-unused-imports "$@";
    printf "\n%10s Checking against 'flake8'\n";
    flake8 "$@";
    printf "\n%10s Errors from 'pylint'\n";
    pylint --rcfile=.pylintrc "$@"
}
```

File: csi/main.py (hunk 1 above)
Comment (leelavg) [reply csi/main.py]: Sure, will take care before that I've a query and will state in the comment.

**Review by amarts — APPROVED** [review]

**Pull request merged**